lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py

```diff
@@ -256,6 +288,17 @@ def main():
         if 'family' not in module.params or not module.params['family']:
             module.fail_json(msg="To use task definitions, a family must be specified")
 
+        launch_type = module.params['launch_type']
+        if launch_type == 'FARGATE' and ('cpu' not in module.params or not module.params['cpu']):
```

Comment:
These could be required_if in AnsibleModule

```
required_if(['launch_type', 'FARGATE', ['cpu', 'memory']])
```

You'll need your existing network_mode check